File: bindings/android/src/commonMain/kotlin/ai/uldren/loom/facets/meetings.kt
package ai.uldren.loom

expect fun Loom.meetingsImportSnapshot(
    path: String,
    workspace: String,
    inputProfile: String,
    snapshot: ByteArray,
    dryRun: Boolean = false,
    passphrase: String? = null,
    kek: ByteArray? = null,
    authPrincipal: String? = null,
    authPassphrase: String? = null,
): String

expect fun Loom.meetingsSourceRead(
    path: String,
    workspace: String,
    sourceId: String,
    leaf: String,
    passphrase: String? = null,
    kek: ByteArray? = null,
    authPrincipal: String? = null,
    authPassphrase: String? = null,
): ByteArray

File: bindings/react-native/src/facets/meetings.ts
import UldrenLoom from '../NativeUldrenLoom';
import { authArgs, keyArgs } from '../internal';
import type { LoomAuth, LoomKey } from '../internal';

export function meetingsImportSnapshot(
  loomPath: string,
  workspace: string,
  inputProfile: string,
  snapshot: Uint8Array | number[],
  dryRun = false,
  key?: LoomKey,
  auth?: LoomAuth
): Promise<string> {
  const [passphrase, kek] = keyArgs(key);
  const [authPrincipal, authPassphrase] = authArgs(auth);
  return UldrenLoom.meetingsImportSnapshot(
    loomPath,
    workspace,
    inputProfile,
    Array.from(snapshot),
    dryRun,
    passphrase,
    kek,
    authPrincipal,
    authPassphrase
  );
}

export async function meetingsSourceRead(
  loomPath: string,
  workspace: string,
  sourceId: string,
  leaf: string,
  key?: LoomKey,
  auth?: LoomAuth
): Promise<Uint8Array> {
  const [passphrase, kek] = keyArgs(key);
  const [authPrincipal, authPassphrase] = authArgs(auth);
  const bytes = await UldrenLoom.meetingsSourceRead(
    loomPath,
    workspace,
    sourceId,
    leaf,
    passphrase,
    kek,
    authPrincipal,
    authPassphrase
  );
  return Uint8Array.from(bytes);
}
